lib/stat_tracker.rb

```diff
@@ -95,6 +95,15 @@ def goals_per_game_per_team
       result[game_team.team_id] << game_team.goals
       result
     end
+  def total_shots_per_team
+    @game_teams.reduce(Hash.new(0)) do |result, game_team|
+      result[game_team.team_id] += game_team.shots
+      result
+    end
+  end
+
+  def shots_to_goals_ratio_per_team
+    total_shots_per_team.merge(total_goals_per_team){|key, oldval, newval| (oldval.to_f / newval.to_f).round(2)}
   end
```

Comment:
Really cool use of merge! I had not seen that before. I think we can name the block variables something more descriptive `|team_id, shots, goals|`

Also within the division you only need to make one value a float then the whole think will evaluate to a float.

Reply:
Have changed it so that only one is a float now!

Reply:
Oh interesting! I've never seen this before either. Cool!

Reply:
That is interesting! Never used merge before, either, but very cool to see!

Reply:
Yup, I remembered seeing a "merge" command during my many scans of ruby-docs and I thought it would be cool to use since I was working with two different hashes. Turns out, it's good!